### PR TITLE
`Project` command in CLI

### PR DESCRIPTION
**Pull request by jamestouri**
First command is list to see the different projects 

ran locally `node dist/cjs/index.cjs project list` and outputted 

```

Candid <> Medplum Demo (Project/7b01521f-4d9e-41c9-8d0e-d3514a84fffe)

Super Admin (Project/ae4eb181-c468-4d20-8f04-4c6b988790dc)

My-first-project (Project/749bdcee-c77f-4fe8-8c3d-8abc44268215)
```

https://github.com/medplum/medplum/issues/1872

### CODE REVIEW

File: packages/cli/src/projects.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Command } from 'commander';
+import { medplum } from '.';
+import { MedplumClient } from '@medplum/core';
+// import { prettyPrint } from './utils';
```
Comment (tonylee80) [inline packages/cli/src/projects.ts:4]: extra comment

**Review by tonylee80 — APPROVED** [review]
Looks good! I love how we can easily add new commands.

File: packages/cli/src/projects.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+function projectList(medplum: MedplumClient): void {
+  const logins = medplum.getLogins();
+
+  const projects = logins
+    .map((login) => {
+      login.project;
+      const keyValue = Object.entries(login.project);
+      const values = keyValue.map(([key, value]) => `${key}: ${value}`);
```
Comment (codyebberson) [inline packages/cli/src/projects.ts:22]: This makes assumptions about the contents of `getLogins()` which might not always be true.

Rather than showing all values, let's be explicit about showing just the project details.

For example:

```ts
.map((login) => `${login.project.display} (${login.project.reference})`)
```

**Review by codyebberson — COMMENTED** [review]
Nice!  One cleanup tweak, otherwise lgtm 👍

File: package-lock.json [hunk 3]
```diff
@@ -16396,6 +16396,7 @@
       "version": "1.1.11",
       "resolved": "https://registry.npmjs.org/brace-expansion/-/brace-expansion-1.1.11.tgz",
       "integrity": "sha512-iCuPHDFgrHX7H2vEI/5xpz07zSHB00TpugqhmYtVmMO6518mCuRMoOYFldEBl0g187ufozdaHgWKcYFb61qGiA==",
+      "dev": true,
```
Comment (tonylee80) [inline package-lock.json:16399]: Just to confirm, is this change intentional?

File: packages/mock/src/client.ts [hunk 4]
```diff
@@ -142,6 +142,13 @@ export class MockClient extends MedplumClient {
     return super.getActiveLogin();
   }
 
+  getLogins(): LoginState[] {
+    if (this.activeLoginOverride !== undefined) {
```
Comment (tonylee80) [inline packages/mock/src/client.ts:146]: What's the uses for this change for the CLI update?

File: package-lock.json (hunk 3 above)
Comment (codyebberson) [reply package-lock.json:16399]: Yeah, we should only change `package-lock.json` if one of the `package.json` files changed.  Let's revert this.

File: packages/mock/src/client.ts (hunk 4 above)
Comment (codyebberson) [reply packages/mock/src/client.ts:146]: This is an oddity of `MockClient`.  By default, `MockClient` is logged in a "Dr. Alice Smith", which is a default `Practitioner` resource that you can use in tests.  `activeLoginOverride` is a `MockClient` feature to modify the current mocked login state.

File: package-lock.json (hunk 3 above)
Comment (jamestouri) [reply package-lock.json:16399]: I ran npm i in the server package to fix a type error in outcomes.ts

```
function getValidationErrorExpression(error: ValidationError): string[] | undefined {
  // ValidationError can be AlternativeValidationError | GroupedAlternativeValidationError | UnknownFieldsError | FieldValidationError
  if (error.type === 'field') {
    return [error.path];
  }
  return undefined;
}
```
The error.type and error.path kept producing errors.

**Review by tonylee80 — APPROVED** [review]
